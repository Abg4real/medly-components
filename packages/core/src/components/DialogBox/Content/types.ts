
export interface StyledProps {
    headerHeight: number;
}
