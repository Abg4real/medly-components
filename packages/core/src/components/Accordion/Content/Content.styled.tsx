import { styled } from '@medly-components/utils';

export const Wrapper = styled('div')`
    cursor: pointer;
    display: flex;
    outline: none;
    border-top: solid 0.2rem ${({ theme }) => theme.accordion.separatorColor};
    padding: 1.8rem;
    background-color: transparent;
`;
