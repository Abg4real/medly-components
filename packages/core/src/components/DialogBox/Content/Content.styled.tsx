import { styled } from '@medly-components/utils';

export const Content = styled('div')`
    flex: 1;
    box-sizing: border-box;

    @media (min-width: 768px) {
        overflow: auto;

        &::-webkit-scrollbar {
            width: 0.5rem;
        }
        &::-webkit-scrollbar-track {
            background-color: transparent;
        }
        &::-webkit-scrollbar-thumb {
            border-radius: 1rem;
            background-color: ${({ theme }) => theme.modal.scrollbarThumbColor};
        }
    }
`;
