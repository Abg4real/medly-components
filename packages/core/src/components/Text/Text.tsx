import { WithStyle } from '@medly-components/utils';
import React, { SFC, useMemo } from 'react';
import { TextStyled } from './Text.styled';
import { Props } from './types';

export const Text: SFC<Props> & WithStyle = React.memo(
    React.forwardRef(({ as, textVariant, textWeight, children, ...restProps }, ref) => {
        const componentType = useMemo(() => {
            let type: keyof JSX.IntrinsicElements | React.ComponentType<any> = 'span';

            if (textVariant === 'body1' || textVariant == 'body2') {
                type = textWeight === 'Strong' || textWeight === 'ExtraStrong' ? 'strong' : 'span';
            } else {
                type = textVariant;
            }

            return type;
        }, [textWeight, textVariant]);
        return (
            <TextStyled {...{ textVariant, textWeight, ...restProps }} as={as || componentType} ref={ref}>
                {children}
            </TextStyled>
        );
    })
);

Text.defaultProps = {
    lineThrough: false,
    uppercase: false,
    fullWidth: false,
    textColor: 'inherit'
};
Text.displayName = 'Text';
Text.Style = TextStyled;
