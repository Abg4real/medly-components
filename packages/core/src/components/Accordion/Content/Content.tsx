import React from 'react';
import { Wrapper } from './Content.styled';

export const Content = React.memo(({ children }) => {
    return <Wrapper>{children}</Wrapper>;
});
Content.displayName = 'Content';
