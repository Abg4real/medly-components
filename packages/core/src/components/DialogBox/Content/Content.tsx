import { WithStyle } from '@medly-components/utils';
import React, { useContext } from 'react';
import { DialogBoxContext } from '../DialogBox.context';
import * as Styled from './Content.styled';

export const Content: React.FC & WithStyle = React.memo(({ children }) => {
    const { id, headerHeight } = useContext(DialogBoxContext);

    return (
        <Styled.Content  {...{ headerHeight, id: `${id}-content` }}>
            {children}
        </Styled.Content>
    );
});
Content.displayName = 'Content';
Content.Style = Styled.Content;
