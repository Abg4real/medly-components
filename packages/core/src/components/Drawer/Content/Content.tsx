import { WithStyle } from '@medly-components/utils';
import React, { useContext, useRef } from 'react';
import { useScrollState } from '../../Modal/useScrollState';
import { DrawerContext } from '../Drawer.context';
import { ContentStyled } from './Content.styled';

export const Content: React.FC & WithStyle = React.memo(props => {
    const contentRef = useRef(),
        { dispatch, scrollState } = useContext(DrawerContext),
        handleScroll = useScrollState({ ref: contentRef, scrollState, dispatch });

    return <ContentStyled ref={contentRef} onScroll={handleScroll} {...props} />;
});

Content.displayName = 'DrawerContent';
Content.Style = ContentStyled;
