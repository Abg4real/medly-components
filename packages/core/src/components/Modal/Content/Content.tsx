import { WithStyle } from '@medly-components/utils';
import React from 'react';
import * as Styled from './Content.styled';
import { Props } from './types';

export const Content: React.FC<Props> & WithStyle = React.memo(({ id, scrollState, dispatch, ...restProps }) => {
    return <Styled.Content {...{ ...restProps, id: `${id}-content` }} />;
});
Content.displayName = 'Content';
Content.Style = Styled.Content;
