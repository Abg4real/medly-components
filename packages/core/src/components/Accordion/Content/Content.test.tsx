import { render } from '@test-utils';
import React from 'react';
import Text from '../../Text';
import { Content } from './Content';

describe('Content component', () => {
    const renderTitle = () =>
        render(
            <Content>
                <Text>Hello</Text>
            </Content>
        );
    it('should render properly', () => {
        const { container } = renderTitle();
        expect(container).toMatchSnapshot();
    });
});
