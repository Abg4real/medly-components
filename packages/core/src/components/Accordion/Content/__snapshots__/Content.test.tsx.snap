// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component should render properly 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c0 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  border-top: solid 0.2rem #eff2f4;
  padding: 1.8rem;
  background-color: transparent;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Hello
    </span>
  </div>
</div>
`;
